diff --git a/node_modules/@ethersproject/web/lib.esm/geturl.js b/node_modules/@ethersproject/web/lib.esm/geturl.js
index 67395d0..3ca4bb3 100644
--- a/node_modules/@ethersproject/web/lib.esm/geturl.js
+++ b/node_modules/@ethersproject/web/lib.esm/geturl.js
@@ -19,7 +19,7 @@ export function getUrl(href, options) {
             headers: (options.headers || {}),
             body: (options.body || undefined),
         };
-        if (options.skipFetchSetup !== true) {
+        if (false) {
             request.mode = "cors"; // no-cors, cors, *same-origin
             request.cache = "no-cache"; // *default, no-cache, reload, force-cache, only-if-cached
             request.credentials = "same-origin"; // include, *same-origin, omit
diff --git a/node_modules/@ethersproject/web/lib/browser-geturl.js b/node_modules/@ethersproject/web/lib/browser-geturl.js
index 7ef6c6e..371e565 100644
--- a/node_modules/@ethersproject/web/lib/browser-geturl.js
+++ b/node_modules/@ethersproject/web/lib/browser-geturl.js
@@ -52,7 +52,7 @@ function getUrl(href, options) {
                         headers: (options.headers || {}),
                         body: (options.body || undefined),
                     };
-                    if (options.skipFetchSetup !== true) {
+                    if (false) {
                         request.mode = "cors"; // no-cors, cors, *same-origin
                         request.cache = "no-cache"; // *default, no-cache, reload, force-cache, only-if-cached
                         request.credentials = "same-origin"; // include, *same-origin, omit
diff --git a/node_modules/@ethersproject/web/src.ts/browser-geturl.ts b/node_modules/@ethersproject/web/src.ts/browser-geturl.ts
index 88d5424..c78922c 100644
--- a/node_modules/@ethersproject/web/src.ts/browser-geturl.ts
+++ b/node_modules/@ethersproject/web/src.ts/browser-geturl.ts
@@ -15,7 +15,7 @@ export async function getUrl(href: string, options?: Options): Promise<GetUrlRes
         body: (options.body || undefined),
     };
 
-    if (options.skipFetchSetup !== true) {
+    if (false) {
         request.mode = <RequestMode>"cors";              // no-cors, cors, *same-origin
         request.cache = <RequestCache>"no-cache";        // *default, no-cache, reload, force-cache, only-if-cached
         request.credentials = <RequestCredentials>"same-origin";  // include, *same-origin, omit
