diff --git a/node_modules/@ethersproject/bignumber/lib/bignumber.d.ts b/node_modules/@ethersproject/bignumber/lib/bignumber.d.ts
index 4864fab..42b4079 100644
--- a/node_modules/@ethersproject/bignumber/lib/bignumber.d.ts
+++ b/node_modules/@ethersproject/bignumber/lib/bignumber.d.ts
@@ -32,7 +32,7 @@ export declare class BigNumber implements Hexable {
     toString(): string;
     toHexString(): string;
     toJSON(key?: string): any;
-    static from(value: any): BigNumber;
+    static from(value: BigNumberish): BigNumber;
     static isBigNumber(value: any): value is BigNumber;
 }
 export declare function _base36To16(value: string): string;
diff --git a/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts b/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts
index 31d4c44..28cd65e 100644
--- a/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts
+++ b/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts
@@ -223,7 +223,7 @@ export class BigNumber implements Hexable {
         return { type: "BigNumber", hex: this.toHexString() };
     }
 
-    static from(value: any): BigNumber {
+    static from(value: BigNumberish): BigNumber {
         if (value instanceof BigNumber) { return value; }
 
         if (typeof(value) === "string") {
